### DISCUSSION TITLE
Github copilot is giving Error sending telemetry FetchError: read ECONNRESET -4077

### DISCUSSION THREAD

**Discussion by trilebh** [community/community#121812] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Github copilot is giving error below. I think the API copilot is down (https://api.githubcopilot.com 404 page not found) and i am getting error. Please help!

[INFO] [default] Error sending telemetry FetchError: read ECONNRESET
at fetch (C:\snapshot\copilot-client\node_modules@adobe\helix-fetch\src\fetch\index.js:99:11)
at processTicksAndRejections (node:internal/process/task_queues:95:5)
at cachingFetch (C:\snapshot\copilot-client\node_modules@adobe\helix-fetch\src\fetch\index.js:288:16)
at cX.fetch (C:\snapshot\copilot-client\lib\src\network\helix.ts:78:22) {
type: 'system',
_name: 'FetchError',
code: 'ECONNRESET',
errno: -4077,
erroredSysCall: 'read'

**Comment by Akash1134** [comment]
Hi @trilebh ,

If you are connecting to Copilot via a proxy, make sure the proxy is configured correctly in your environment. For more information, see "[Configuring network settings for GitHub Copilot](https://docs.github.com/en/copilot/configuring-github-copilot/configuring-network-settings-for-github-copilot#configuring-proxy-settings-for-github-copilot)."

GitHub Copilot uses custom code to connect to proxies. This means a proxy setup supported by your editor is not necessarily supported by GitHub Copilot. Some common causes for errors related to proxies are:

- If your proxy's URL starts https://, it is not currently supported by GitHub Copilot.
- You may need to authenticate to the proxy. GitHub Copilot supports basic authentication or authentication with Kerberos. If you are using Kerberos, ensure you have a valid ticket for the proxy service and that you are using the correct service principal name for the service. For more information, see "[Configuring network settings for GitHub Copilot](https://docs.github.com/en/copilot/configuring-github-copilot/configuring-network-settings-for-github-copilot#authentication-with-kerberos)."
- GitHub Copilot may reject custom certificates. For more information, see "[Troubleshooting certificate-related errors](https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-network-errors-for-github-copilot#troubleshooting-certificate-related-errors)

Thanks!

**Comment by taised** [comment]
I have the same error but i'm not using a proxy. I restarted VS code, it says it removed some certificates, for a short while copilot started working again but then it stopped. I don't think i can restart VS code every half hour to make it work properly. Any other solution to overcome the problem? Thanks

**Reply by yavuzergun1** [reply]
Could you find any solution regarding this issue?

**Comment by austinbenz99** [comment]
It looks like a connection issue, possibly on GitHub's end. You might try restarting VS Code, checking your network, or waiting a bit in case it's a temporary outage.

**Comment by sinkr** [comment]
You will also receive this error if you have aggressive DNS-level filtering for privacy, ads, etc.
